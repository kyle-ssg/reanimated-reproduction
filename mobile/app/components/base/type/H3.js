import React, { Component } from 'react';
import propTypes from 'prop-types';

const h3 = class extends Component {
    static displayName = 'H3';

    render() {
        return (
            <Text
              accessible={this.props.accessible}
              accessibilityLabel={this.props.accessibilityLabel}
              style={[Styles.h3, this.props.style]}
            >
                {this.props.children}
            </Text>
        );
    }
};

h3.propTypes = {
    accessible: propTypes.bool,
    accessibilityLabel: propTypes.string,
    style: propTypes.any,
    children: propTypes.node,
};

module.exports = h3;
