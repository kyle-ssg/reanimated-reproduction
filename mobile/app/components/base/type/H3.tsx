import { FC, ReactNode } from 'react'
import { TextStyle } from 'react-native'

interface Props {
  accessible?: boolean
  accessibilityLabel?: string
  style?: TextStyle | TextStyle[]
  children?: ReactNode
}

const h3: FC<Props> = ({ accessible, accessibilityLabel, style, children }) => (
  <Text
    accessible={accessible}
    accessibilityLabel={accessibilityLabel}
    style={[Styles.h3, style]}
  >
    {children}
  </Text>
)

export default h3
