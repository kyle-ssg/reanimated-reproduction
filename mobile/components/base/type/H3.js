/**
 * Created by kylejohnson on 11/05/2016.
 */
/**
 * Created by kylejohnson on 04/05/2016.
 */
const h3 = class extends React.Component {
	render() {
		return (
			<Text style={[Styles.h3, this.props.style]}>{this.props.children}</Text>
		);
	}
};

h3.propTypes = {
	style: React.PropTypes.any,
	children: OptionalElement
};

module.exports = h3;
