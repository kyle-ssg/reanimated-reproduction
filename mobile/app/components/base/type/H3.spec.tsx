import * as React from "react";
import H3 from "./H3";
import renderer from "react-test-renderer";


describe("<H3 />", () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<H3>Example Title</H3>);
    expect(tree).toMatchSnapshot();
  });
});
