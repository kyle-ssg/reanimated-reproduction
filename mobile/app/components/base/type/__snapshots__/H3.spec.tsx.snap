// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H3 /> renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "System",
        "fontSize": 20,
      },
      undefined,
    ]
  }
>
  Example Title
</Text>
`;
